File: app/src/main/java/tw/com/louis383/coffeefinder/di/AppComponent.java
package tw.com.louis383.coffeefinder.di;

import javax.inject.Singleton;

import dagger.Component;
import tw.com.louis383.coffeefinder.details.DetailsFragment;
import tw.com.louis383.coffeefinder.list.ListFragment;
import tw.com.louis383.coffeefinder.mainpage.MainActivity;
import tw.com.louis383.coffeefinder.maps.MapsFragment;
import tw.com.louis383.coffeefinder.model.CoffeeTripAPI;
import tw.com.louis383.coffeefinder.model.PreferenceManager;

/**
 * Created by louis383 on 2017/2/22.
 */

@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {
    CoffeeTripAPI getCoffeeTripAPI();
    PreferenceManager getPreferenceManager();

    void inject(MainActivity mainActivity);

    void inject(MapsFragment mapsFragment);
    void inject(DetailsFragment detailsFragment);
    void inject(ListFragment listFragment);
}

File: app/src/main/java/tw/com/louis383/coffeefinder/di/components/AppComponent.kt
package tw.com.louis383.coffeefinder.di.components

import dagger.Component
import javax.inject.Singleton
import tw.com.louis383.coffeefinder.di.module.AppModule
import tw.com.louis383.coffeefinder.mainpage.MainActivity
import tw.com.louis383.coffeefinder.maps.MapsFragment
import tw.com.louis383.coffeefinder.details.DetailsFragment
import tw.com.louis383.coffeefinder.di.module.LocationModule
import tw.com.louis383.coffeefinder.list.ListFragment

/**
 * Created by louis383 on 2017/2/22.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class, LocationModule::class))
interface AppComponent {
    fun inject(mainActivity: MainActivity)
    fun inject(mapsFragment: MapsFragment)
    fun inject(detailsFragment: DetailsFragment)
    fun inject(listFragment: ListFragment)
}